File: test/elixir/support/fixtures/ex_js_consistency/erlang/module_3.ex
# credo:disable-for-this-file Credo.Check.Readability.Specs
defmodule Hologram.Test.Fixtures.ExJsConsistency.Erlang.Module3 do
  @moduledoc """
  Custom :error_info format module used by :erlang.error/3 consistency tests.
  """

  def format_error(_reason, [{_module, _function, _args, _info} | _rest]) do
    %{2 => "not a map"}
  end
end

File: test/javascript/support/fixtures/ex_js_consistency/erlang/module_3.mjs
"use strict";

import Interpreter from "../../../../../../assets/js/interpreter.mjs";
import Type from "../../../../../../assets/js/type.mjs";

export function defineModule3Fixture() {
  Interpreter.defineElixirFunction(
    "Hologram.Test.Fixtures.ExJsConsistency.Erlang.Module3",
    "format_error",
    2,
    "public",
    [
      {
        params: (_context) => [
          Type.variablePattern("_reason"),
          Type.variablePattern("_stacktrace"),
        ],
        guards: [],
        body: (_context) => {
          return Type.map([[Type.integer(2), Type.bitstring("not a map")]]);
        },
      },
    ],
  );
}
